docs/testing/releases/330.md

```diff
@@ -0,0 +1,51 @@
+## Testing notes and ZIP for release 3.3.0
+
+[![Create Todo list](https://raw.githubusercontent.com/senadir/todo-my-markdown/master/public/github-button.svg?sanitize=true)](https://git-todo.netlify.app/create)
+
+Zip file for testing: [woocommerce-gutenberg-products-block.zip](https://github.com/woocommerce/woocommerce-gutenberg-products-block/files/5148289/woocommerce-gutenberg-products-block.zip)
+
+### Enhancements – Cart & Checkout   <!-- heading -->
+
+#### Express payment methods in Cart block (Apple Pay, Chrome Pay)  <!-- heading -->
+
+- [ ] This feature requires the following:
+  - [WooCommerce Stripe extension](https://woocommerce.com/products/stripe) installed and active.
+  - One or more express payment methods supported and enabled on the store and client system (OS, browser). Note that Chrome Pay and Apple Pay are not supported in all regions. Please refer to documentation for Apple Pay and Chrome Pay for more information.
+- [ ] Go to `Dashboard > WooCommerce > Settings > Payments` and activate `Stripe – Credit Card`.
+- [ ] Click `Manage` button for `Stripe – Credit Card`.
+- [ ] Ensure `Payment Request Buttons` checkbox is enabled, and click `Save changes`.
+- [ ] Add the Cart block to a page and publish the page (for example, the main store `cart` page).
+- [ ] In the editor, the Cart block should show a preview of express payment options (for example, Apple Pay button).
+- [ ] View store front end in a browser with an express payment method enabled and configured. For example Chrome or Safari.
+- [ ] Add some products to cart.
+- [ ] View cart. If express payment is supported and available, you should see relevant express payment buttons above the cart `Proceed to checkout` button.
+  - Note: if you don't see any express payment options, confirm that they are working in the Checkout block, product page, or shortcode checkout page. 
+  - The cart block should include the same express payment options as other store pages.
+
+### Bug Fixes – Cart & Checkout   <!-- heading -->
+
+#### Fix alignment of discounted prices in Cart block - sale prices right-aligned  <!-- heading -->
+
+- [ ] Add a product on sale to the Cart and visit the Cart block.
+- [ ] Verify regular and discounted prices are both aligned to the right (see screenshots above for comparison).
+
+#### Fix alignment of discounted prices in Cart block - sale price alignment options in product price block  <!-- heading -->
```

Comment:
Product price block is behind a feature flag, should we exclude this item from the testing notes?

Reply:
It is behind a feature flag, not an experimental flag; we shipped this several releases ago.
Pca54o-7j-p2
https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2860
it seems we didn't triage it to a milestone. It should have been part of 3.1.0.

Reply:
I _think_ all the product element blocks are registered normally, i.e. not behind a feature flag! (I was surprised, I assumed these were behind flag.)

@senadir can you confirm I understand this correctly? 

- There are [three block registration functions](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/main/assets/js/settings/blocks/feature-flags.js) - `registerBlockType` from core, and `registerFeaturePluginBlockType` / `registerExperimentalBlockType`.
- Each block registers itself as normal (available everywhere), available in feature plugin, or available in dev build (experimental) by calling one of these functions to register.
- All [product element blocks](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/main/assets/js/atomic/blocks/product-elements) use `registerBlockType`, so are not behind feature flag - they are registered for woo core and for feature plugin.
- However, product element blocks [specify a parent block type](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/main/assets/js/atomic/blocks/product-elements/shared/config.js) (unless experimental/dev build) - currently `All Products` and `Single Product`. This means that they are only available within that block context.
- [`Single Product` registers itself as experimental](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/main/assets/js/blocks/single-product/index.js) - so is only available in dev build.
- Therefore - the atomic product blocks are fully available in `All Products` block in woo core and in release plugin.

Thanks for prompting this @Aljullu  – I wasn't clear on this!

I'll update the testing docs with this in mind 😄

Reply:
OK I've now tested this and I found an issue. The price alignment options don't work in the front end or editor grid preview. They only work in the editor, when editing the layout (as far as I can tell, in my testing). 

I think I'll reopen that issue for now - #3049, and we can decide what action to take with the release (e.g. fix, release with known issue, or revert fix).

Reply:
> All product element blocks use registerBlockType, so are not behind feature flag - they are registered for woo core and for feature plugin.

Not all, previously available atomic blocks are behind `registerBlockType`, those are available in All Products, new blocks for Single Product are behind `registerExperimentalBlockType`, those are:
⚛️ Add to cart
⚛️ Product category list
⚛️ Product SKU
⚛️ Product stock indicator
⚛️ Product tag list

> However, product element blocks specify a parent block type (unless experimental/dev build) - currently All Products and Single Product. This means that they are only available within that block context.

Yes, if you're in a dev instance, you will them under `⚛️ Element Blocks` in the inserter, if you're in a normal build, you won't see them and can only insert them inside All Products.

Reply:
Ah I see, you're correct @senadir - I didn't check them all!

That makes sense – some element blocks are intended to be available in `All Products` with no feature flag limitation: `Title, Summary, Sale Badge, Rating, Price, Image, Button`.  

The new feature that is in 3.3.0 is alignment and colour options for some of these blocks. `Product Price` alignment option was logged as an issue/PR in the milestone (#3049). I'll investigate to confirm what else changed and add to changelog/testing instructions.